lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -1,16 +1,16 @@
 module Msf::Payload::Adapter::Fetch
-
   def initialize(*args)
     super
     register_options(
       [
         Msf::OptBool.new('FETCH_DELETE', [true, 'Attempt to delete the binary after execution', false]),
-        Msf::OptString.new('FETCH_FILENAME', [ false, 'Name to use on remote system when storing payload; cannot contain spaces or slashes', Rex::Text.rand_text_alpha(rand(8..12))], regex: /^[^\s\/\\]*$/),
+        Msf::OptBool.new('FETCH_FILELESS', [true, 'Attempt to run payload without touching disk, Linux ≥3.17 only', false]),
```

Comment:
Now that I think about it, this probably should go in the linux_options file

Reply:
After a talk with Spencer, I think we should put this in the linux_options file and move the `WRITABLE_DIR` declaration to linux_options and windows_options rather than just setting the default value there.  That way, we can make the `WRITABLE_DIR` option display only if `FETCH_FILELESS` is set to false.